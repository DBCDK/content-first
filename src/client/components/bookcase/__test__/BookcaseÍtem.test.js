import React from 'react';
import renderer from 'react-test-renderer';

import {BookcaseItem} from '../BookcaseItem.component';

jest.mock(
  '../../general/CheckmarkConnected.component',
  () => 'CheckmarkConnected'
);
jest.mock('../../pulse/Pulse.component', () => 'Pulse');
jest.mock('../CarouselItem.component', () => 'CarouselItem');
jest.mock('../CarouselSlider.component', () => 'CarouselSlider');

const celeb = {
  id: '1',
  name: 'Bjarne Slot Christiansen',
  img: 'img/bookcase/BS3.png',
  description: 'lorem ipsum',
  bookcase: 'img/bookcase/BS-bogreol.png',
  books: [
    {
      pid: '870970-basis:52530423',
      position: {x: 13.5, y: 40},
      description: 'lorem ipsum . . .'
    },
    {
      pid: '870970-basis:53079202',
      position: {x: 21.8, y: 44},
      description: 'lorem ipsum . . .'
    }
  ]
};

const book = [
  {
    book: {
      pid: '1234',
      creator: 'Nicole Boyle RÃ¸dtnes',
      title: 'test',
      cover: 'img/cover.png',
      description: 'lorem ipsum...'
    }
  }
];

describe('BookcaseItem', () => {
  it('renders initial component', () => {
    const tree = renderer
      .create(<BookcaseItem celeb={celeb} books={book} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
