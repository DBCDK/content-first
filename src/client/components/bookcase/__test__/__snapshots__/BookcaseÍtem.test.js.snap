// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookcaseItem renders initial component 1`] = `
<section
  className=""
  onClick={undefined}
>
  <img
    alt="undefined bogreol"
    src="img/bookcase/BS-bogreol.png"
  />
  <div
    className="row"
  >
    <div
      className="col-xs-4 celeb"
    >
      <img
        alt="luk"
        className="carousel-close"
        onClick={[Function]}
        src="/static/media/Kryds.e69a54ef.svg"
      />
      <div
        className="col-xs-12 celeb-top"
      >
        <div
          className="col-xs-12 celeb-img"
        >
          <img
            alt="Bjarne Slot Christiansen"
            src="img/bookcase/BS3.png"
          />
        </div>
        <div
          className="col-xs-12 celeb-title"
        >
          <h1>
            Bjarne Slot Christiansen
          </h1>
        </div>
        <div
          className="col-xs-12 celeb-description"
        >
          <p>
            lorem ipsum
          </p>
          <button
            className="celeb-books-btn btn btn-info"
            onClick={[Function]}
            type="button"
          >
            Se bøger
          </button>
        </div>
      </div>
      <div
        className="col-xs-12 celeb-bottom"
      >
        <CarouselSlider
          onNextBook={[Function]}
          slideIndex={null}
        >
          <CarouselItem
            active={false}
            book={
              Object {
                "cover": "img/cover.png",
                "creator": "Nicole Boyle Rødtnes",
                "description": "lorem ipsum...",
                "pid": "1234",
                "title": "test",
              }
            }
            description="lorem ipsum..."
          />
        </CarouselSlider>
      </div>
    </div>
    <div
      className="col-xs-8 bookswrap"
    >
      <Pulse
        active=""
        onClick={[Function]}
        pid="870970-basis:52530423"
        position={
          Object {
            "x": 13.5,
            "y": 40,
          }
        }
      />
      <Pulse
        active=""
        onClick={[Function]}
        pid="870970-basis:53079202"
        position={
          Object {
            "x": 21.8,
            "y": 44,
          }
        }
      />
    </div>
  </div>
</section>
`;
